sa/database_test.go

```diff
@@ -73,3 +73,26 @@ func TestStrictness(t *testing.T) {
 		t.Fatalf("Got wrong type of error: %s", err)
 	}
 }
+
+func TestSafeSelectOne(t *testing.T) {
```

Comment:
My original justification for not doing this was that other tests in `sa.go` that call this are tested, so this still gets full coverage.

I'm not really sure doing a full mock for `gorp.DbMap` (although I guess we could just do it by passing the method `dbMap.SelectOne` instead) is really worth it given that, do you feel strongly about this?